frame/support/procedural/src/storage/parse.rs

```diff
@@ -38,10 +38,36 @@ mod keyword {
 	syn::custom_keyword!(hasher);
 }
 
+/// Specific `Opt` to implement structure with optional parsing
+#[derive(Debug, Clone)]
+pub struct Opt<P> {
+	pub inner: Option<P>,
+}
```

Comment:
This structure implements Parse similarly to how Parse is implemented for `Option<WhereClause>`, we check a token and if this token exist we parse the inner type otherwise we return None